lib/ContainerManager.js

```diff
@@ -521,12 +619,62 @@ Server.prototype.routes.post['container/proxy'] = function(req, res, next) {
 }
 
 /**
- * List builds on this server.
+ * List builds on this server that have a matching existing container (by default).
+ * Use ?all=true query param to get all build records.
+ * NOTE: does NOT do a live check for an existing container. Relies on meta-data that's in the build.container object.
  */
 Server.prototype.routes.get['builds'] = function(req, res) {
   var readStream = this.streamFromDB('builds')
   var jsonStream  = new JSONStream.stringify()
-  readStream.pipe(jsonStream).pipe(res);
+  var buildsStream = readStream
+
+  if(req.query.all !== "true"){
+    // filter out builds that don't have an existing container by default
+    buildsStream = buildsStream.pipe(through2.obj(function(build, enc, callback){
+      if(build.container && build.container.state){
+        callback(null, build)
+      } else {
+        callback()
+      }
+    }))
+  }
+
+  buildsStream.pipe(jsonStream).pipe(res);
+};
+
+/**
+ * Update the container status for all builds. Intended as a one-time catchup to
+ * add metadata set by updateBuildContainerStatus()
+ */
+Server.prototype.routes.post['updatebuilds'] = function(req, res) {
```

Comment:
URL paths should have `-` between words. This is consistent with `startbuild` but we should change that one as well.